Fix CMake configuration to make Physsim usable as a library

tmadlener opened the pull request:
BEGINRELEASENOTES
- Update CMake configuration to use the modern target based approach of propagating dependencies and making Physsim a package that can be easily found via `find_package(Physsim)`
- Remove duplicated headers in `src` and use the ROOT CMake macro to generate dictionaries and build the library.
- Make examples use this functionality and in this way make it build again via the canonical iLCSoft cmake build steps.
  - Add more detailed instructions to the README
- Add github actions based CI and remove the travis-ci configuration.

ENDRELEASENOTES

tmadlener committed:
Add basic .gitignore

tmadlener committed:
Use modern cmake approach to get downstream usability

Define targets properly and export and install them such, that they are
easily usable with find_package

tmadlener committed:
Make examples build again

tjunping reviewed (approved):
looks fine to me

tmadlener committed:
Add github actions CI workflows

tmadlener committed:
Remove travis-ci configuration

JennyListDESY commented:
When building against ilcsoft v02-02-03, 
an additional 
INCLUDE_DIRECTORIES( ./include/physsim )
is needed in CMakeLists.txt

tmadlener committed:
Cleanup dictionary generation and libray building

- Remove duplicated headers
- Fix include paths
- Merge LinkDef.h files into one and remove individual ones
- Explicitly state library source files
- Use CMake macro provided by root

tmadlener committed:
Update README with build instructions

tmadlener commented:
After a bit of fiddling and some more cleanup, this now works with key4hep and iLCSoft releases (at least with recent ones), and the CI workflows now also build the example processors. In the course of the cleanup I have removed the duplicated headers in the `src/*` directories and made sure that all the necessary include paths are set correctly. It should now be rather straight forward to move the example processors into a separate repository, but I would leave that for another pull request.

tmadlener commented:
Since all the testing has been done successfully, I think we can merge this and fix any upcoming issues in a separate pull request to not make this too cluttered.

andresailer commented on CMakeLists.txt:
```diff
@@ -14,11 +13,11 @@ SET( ${PROJECT_NAME}_VERSION_MAJOR 0 )
-cmake_policy(SET CMP0008 NEW)  
+#--- Define basic build settings -----------------------------------------------
+# Provides install directory variables as defined for GNU software
+include(GNUInstallDirs)
```
This makes the libraries now go to lib64, which breaks the check in ilcsoft-install
(And the assignment of the library to MARLIN_DLL, which happens because of how physsim is done in ilcsoft install https://github.com/iLCSoft/iLCInstall/blob/b32418db68960110ade0270a7ce37fbe32241983/releases/HEAD/release-ilcsoft.cfg#L167-L168 )

tmadlener replied:
That is true. I have made a PR to iLCInstall to accomodate for the `lib[64]` changes. `MARLIN_DLL` should not really matter for Physsim, because it does not build processors. Those need to be explicitly built from the examples directory.
